### PR TITLE
Add `<picture>` and `<source>` elements

### PR DESCRIPTION
**Pull request by petele**
I couldn't find an element for `<source>` so I added it here. The description probably needs a little tweak.

### CODE REVIEW

File: features/picture.yml [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
 name: <picture>
-description: The `<picture>` element TODO.
+description: The `<picture>` element is used for responsive images.
 spec: https://html.spec.whatwg.org/multipage/embedded-content.html#the-picture-element
 group: html
 caniuse: picture
 compat_features:
   - api.HTMLPictureElement
   - html.elements.picture
+  - html.elements.source
```
Comment (foolip) [inline features/picture.yml:9]: The `<source>` element was first introduced for `<video>`, so it should either be part of that feature, or possibly its own feature.

File: features/picture.yml [hunk 2]
```diff
@@ -1,9 +1,9 @@
-draft_date: 2024-06-05
 name: <picture>
-description: The `<picture>` element TODO.
+description: The `<picture>` element is used for responsive images.
```
Comment (foolip) [inline features/picture.yml:2]: I think needs to be elaborated a little bit. https://web.dev/learn/design/picture-element looks like a good intro to the element, maybe a description can be condensed from that.

Comment (petele) [reply features/picture.yml]: Updated

File: features/picture.yml (hunk 1 above)
Comment (petele) [reply features/picture.yml]: Split out into a separate element

**Comment by petele** [comment]
Creates a _separate_ entry for `<source>`. 

On `<source>` I used `compute_from` because the `height` and `width` properties only landed in late 2022. I'm open to putting them into their own feature, but not convinced that's required.

File: features/source.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+name: <source>
+description: The `<source>` element specifies a media resource for the `<picture>`, `<video>`, or `<audio>` elements.
```
Comment (ddbeck) [inline features/source.yml:2]: I like this, but there's a couple of style issues.

First, see [the guidance on _specifies_](https://github.com/web-platform-dx/web-features/blob/main/docs/README.md#specifies.

The current phrasing with "or" also sound ambiguous to me, as if it's not known which of the elements it applies to. One of these alternatives might do.

```suggestion
description: The `<source>` element sets a media resource for an `<picture>`, `<video>`, or `<audio>` element.
```

```suggestion
description: The `<source>` element sets a media resource for the `<picture>`, `<video>`, and `<audio>` elements.
```

Comment (petele) [reply features/source.yml]: done.

**Review by ddbeck — APPROVED** [review]

**Pull request merged**